src/internal.c

```diff
@@ -6597,727 +6708,952 @@ int CopyDecodedToX509(WOLFSSL_X509* x509, DecodedCert* dCert)
 
 #endif /* KEEP_PEER_CERT || SESSION_CERTS */
 
+typedef struct DoCertArgs {
```

Comment:
My preference was to keep the struct that contains the locals for the function right above it. Agree these is value having struct/typedefs at top of file, but in this case with how large the file and that this struct is only used in the DoCertificate and FreeDoCertArgs functions, I'd like to keep these here for now.